Building with `STAT_VM` segfaults...

Hi @egallesio !

I have tried to build STklos with VM statistics, and it segfaulted.

```
./configure CC=gcc CFLAGS="-Wall -Wextra -g -O0 -DSTAT_VM" --enable-threads=none  && make  -j9
```

The result was:

```
./tmpcomp -o stklos-compile stklos-compile.stk
./tmpcomp -o stklos-genlex stklos-genlex.stk
(cd ../lib/stklos; make  preproc.ostk)
make[2]: Entering directory '/home/jeronimo/pkg/scheme/STklos-github/lib/stklos'
../../utils/tmpcomp -o preproc.ostk preproc.stk
Segmentation fault
Segmentation fault
make[1]: *** [Makefile:575: stklos-genlex] Error 139
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:572: stklos-compile] Error 139
Segmentation fault
make[2]: *** [Makefile:659: preproc.ostk] Error 139
make[2]: Leaving directory '/home/jeronimo/pkg/scheme/STklos-github/lib/stklos'
make[1]: *** [Makefile:581: ../lib/stklos/preproc.ostk] Error 2
make[1]: Leaving directory '/home/jeronimo/pkg/scheme/STklos-github/utils'
```

Running `gdb src/stklos` shows this backtrace:

```
Program received signal SIGSEGV, Segmentation fault.
hash_string (string=0x400000004 <error: Cannot access memory at address 0x400000004>) at hash.c:86
86	  for (c = *string; c ; c = *string++) {
(gdb) where
#0  hash_string (string=0x400000004 <error: Cannot access memory at address 0x400000004>) at hash.c:86
#1  0x00005555555828ee in hash_get_variable (h=0x7ffff7f83f58, v=0x7ffff7719960, index=0x7fffffffda6c) at hash.c:354
#2  0x000055555558297e in STk_hash_get_variable (h=0x7ffff7f83f58, v=0x7ffff7719960) at hash.c:365
#3  0x000055555557a4fe in STk_lookup (symbol=0x7ffff7719960, env=0x7ffff7f83f30, ref=0x7fffffffdb70, err_if_unbound=0) at env.c:781
#4  0x00005555555ae877 in run_vm (vm=0x7ffff7f8df00) at vm.c:1095
#5  0x00005555555b3ec9 in STk_boot_from_C () at vm.c:2322
#6  0x00005555555a2849 in main (argc=0, argv=0x7fffffffe020) at stklos.c:276
```

My plan was to add timings to the statistics. I see you have already written code to compute the frequencies of instructions *and* pairs of instructions. I wanted to add a `clock()` call there too, and create a timing table for instructions, but the current code doesn't compile with `STAT_VM`. And I have reviewed the code, but couldn't find anything wrong in it. (?)

--- a/src/vm.c
+++ b/src/vm.c
@@ -55,7 +55,8 @@ static int debug_level = 0;     /* 0 is quiet, 1, 2, ... are more verbose */
 #else
    /* Standard C compiler. Use the classic switch statement */
 #  define CASE(x)       case x:
-#  define NEXT          continue;/* Be sure to not use continue elsewhere */
+#  define NEXT          goto VM_LOOP_END; /* NOT continue, as it interacts badly with the
+                                             do{...}while(0) guards. */
 #endif
 
 #define NEXT0           do{vm->val = STk_void; vm->valc = 0; NEXT;}while(0)
@@ -1547,7 +1548,7 @@ CASE(INSCHEME) {
   NEXT1;
  }
 
- 
+
 CASE(END_OF_CODE) {
    return;
  }
@@ -1709,7 +1710,7 @@ CASE(IN_CXR) {
   vm->val= STk_cxr(vm->val, fetch_const());
   NEXT1;
  }
- 
+
 CASE(IN_APPLY)   {
   STk_panic("INSTRUCTION IN-APPLY!!!!!!!!!!!!!!!!!!!!!!!");
   NEXT;
@@ -1937,7 +1938,8 @@ end_funcall:
       default:
         STk_panic("INSTRUCTION %d NOT IMPLEMENTED\n", byteop);
     }
-  }
+  VM_LOOP_END: /* Where every instruction jumps after executing */
+  } /* for( ; ; ) */
 #endif
   STk_panic("abnormal exit from the VM");
 }

